### PR TITLE
Remove workflow store entries before job store entries.

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

File: engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala [hunk 1]
```diff
@@ -5,20 +5,26 @@ import cromwell.backend.async.JobAlreadyFailedInJobStore
...
-class SqlJobStore(sqlDatabase: JobStoreSqlDatabase) extends JobStore {
+class SqlJobStore(sqlDatabase: SqlDatabase) extends JobStore {
+  val log = LoggerFactory.getLogger(classOf[SqlJobStore])
+
   override def writeToDatabase(workflowCompletions: Seq[WorkflowCompletion], jobCompletions: Seq[JobCompletion], batchSize: Int)(implicit ec: ExecutionContext): Future[Unit] = {
+    val completedWorkflowIds = workflowCompletions.map(_.workflowId.toString)
     for {
       _ <- sqlDatabase.addJobStores(jobCompletions map toDatabase, batchSize)
-      _ <- sqlDatabase.removeJobStores(workflowCompletions.map(_.workflowId.toString))
+      _ <- Future.sequence(completedWorkflowIds map sqlDatabase.removeWorkflowStoreEntry)
```
Comment (danbills) [inline engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala]: you could `traverse` w/ `cats.instances.future._` and `cats.syntax.traverse._`

Comment (mcovarr) [inline engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala]: 😺

**Comment by Horneth** [comment]
👍

**Comment by danbills** [comment]
👍

File: engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala [hunk 2]
```diff
@@ -1,24 +1,34 @@
...
+class SqlJobStore(sqlDatabase: SqlDatabase) extends JobStore {
+  val log = LoggerFactory.getLogger(classOf[SqlJobStore])
+
   override def writeToDatabase(workflowCompletions: Seq[WorkflowCompletion], jobCompletions: Seq[JobCompletion], batchSize: Int)(implicit ec: ExecutionContext): Future[Unit] = {
+    val completedWorkflowIds = workflowCompletions.toList.map(_.workflowId.toString)
     for {
       _ <- sqlDatabase.addJobStores(jobCompletions map toDatabase, batchSize)
-      _ <- sqlDatabase.removeJobStores(workflowCompletions.map(_.workflowId.toString))
+      _ <- (completedWorkflowIds map sqlDatabase.removeWorkflowStoreEntry).sequence[Future, Int]
+      _ <- (completedWorkflowIds map sqlDatabase.removeDockerHashStoreEntries).sequence[Future, Int]
```
Comment (danbills) [inline engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala]: `completedWorkFlowIds traverse sqlDatabase.removeDockerHashStoreEntries` should work

Comment (mcovarr) [inline engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala]: Thanks, that's way nicer!

**Pull request merged**